packages/test/test-service-load/src/loadTestDataStore.ts

```diff
@@ -76,22 +77,26 @@ export class LoadTestDataStoreModel {
 			resolveIfConnectedOrDisposed();
 		});
 
-		const lastKnownSeq = runtime.deltaManager.lastKnownSeqNumber;
+		const deltaManager = toDeltaManagerInternal(runtime.deltaManager);
+		const lastKnownSeq = deltaManager.lastKnownSeqNumber;
 		assert(
-			runtime.deltaManager.lastSequenceNumber <= lastKnownSeq,
+			deltaManager.lastSequenceNumber <= lastKnownSeq,
 			"lastKnownSeqNumber should never be below last processed sequence number",
 		);
 
 		await new Promise<void>((resolve) => {
 			const resolveIfDisposedOrCaughtUp = (op?: ISequencedDocumentMessage) => {
 				if (runtime.disposed || (op !== undefined && lastKnownSeq <= op.sequenceNumber)) {
-					runtime.deltaManager.off("op", resolveIfDisposedOrCaughtUp);
+					toDeltaManagerInternal(runtime.deltaManager).off(
+						"op",
+						resolveIfDisposedOrCaughtUp,
+					);
 					runtime.off("dispose", resolveIfDisposedOrCaughtUp);
 					resolve();
 				}
 			};
 
-			runtime.deltaManager.on("op", resolveIfDisposedOrCaughtUp);
+			toDeltaManagerInternal(runtime.deltaManager).on("op", resolveIfDisposedOrCaughtUp);
```

Comment:
```suggestion
			deltaManager.on("op", resolveIfDisposedOrCaughtUp);
```

Reply:
That causes a difference in behavior if runtime.deltamanager is reassigned between when waitForCatchupOrDispose is called and when the event runs. I know the runtime.deltaManager is reassigned in our mock runtime in some cases (I had a similar issues there which failed tests) so it did this this way to mitigate that risk. If you are confident that wouldn't be an issue here, I could apply this change.

Reply:
in the real implementation we should never be reassigning it

Reply:
actually, I don't think what I did correctly handles the possible issue I mentioned, since its still wrong if its reassigned (the on and of wouldn't match). Applying your suggestion seems like a good option: do you know in real cases the delta runtime.deltamanager  won't be reassigned?

Reply:
Done.